ci/tools/test_encoding.py

```
@@ -194,6 +274,31 @@ def fetch_response(port: int) -> tuple[bytes, str, str]:
+        "/cache-control-no-transform-repeat.js",
+    ):
+        body, encoding, _ = fetch_path(port, path)
+        if encoding:
+            raise RuntimeError(
+                f"{path}: expected no Content-Encoding for no-transform, got {encoding!r}"
+            )
+        if body != expected:
+            raise RuntimeError(f"{path}: identity response body changed")
+
+    if public_body == expected:
+        raise RuntimeError("Cache-Control public negative control was not transformed")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)